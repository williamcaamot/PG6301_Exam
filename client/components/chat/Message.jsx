import React from "react";

function Message(props) {
    return <>
        <div
            style={{
                width: "100%",
                display: "flex",
                flexWrap: "nowrap",
                alignItems:"center",
            }}
        >
            <div>
                <img
                    style={{
                        height:"30px"
                    }}
                    src={props.message.picture}/>
            </div>
            <div><p
                style={{padding:"5px"}}
            >{props.message.nickname}:</p></div> {/*TODO change to nickname*/}


            <div>
                <p>{props.message.message}</p>
            </div>

        </div>


    </>
}

export default Message;