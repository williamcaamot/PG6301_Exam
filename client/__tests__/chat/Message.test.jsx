import renderer, {act} from "react-test-renderer";
import Message from "../../components/chat/Message.jsx";
import {MemoryRouter} from "react-router-dom";

describe("chat message", () => {

    it("show a message", () => {

        const message = {
            message: "daww",
            sender: "williamcoucheronaamot@gmail.com",
            nickname: "WILLIAM123",
            picture: "https://lh3.googleusercontent.com/a/ACg8ocIOgSmXkWTJqoKPe5Hk3JlkkhfUNer4sWkVo3UuIa_dzg=s96-c",
            time: "2023-11-22T13:28:21.025Z"
        }

        const component = renderer.create(
            <MemoryRouter>
                <Message message={message}/>
            </MemoryRouter>);
        expect(component).toMatchSnapshot();
    })


})