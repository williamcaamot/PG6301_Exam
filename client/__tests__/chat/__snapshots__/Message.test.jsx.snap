// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chat message show a message 1`] = `
<div
  style={
    {
      "alignItems": "center",
      "display": "flex",
      "flexWrap": "nowrap",
      "width": "100%",
    }
  }
>
  <div
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "flexWrap": "nowrap",
      }
    }
  >
    <a
      href="/profile/williamcoucheronaamot@gmail.com"
      onClick={[Function]}
      style={
        {
          "display": "flex",
        }
      }
    >
      <div>
        <img
          src="https://lh3.googleusercontent.com/a/ACg8ocIOgSmXkWTJqoKPe5Hk3JlkkhfUNer4sWkVo3UuIa_dzg=s96-c"
          style={
            {
              "height": "30px",
            }
          }
        />
      </div>
      <div>
        <p
          style={
            {
              "padding": "5px",
            }
          }
        >
          WILLIAM123
          :
        </p>
      </div>
    </a>
    <div
      style={
        {
          "paddingBottom": "5px",
        }
      }
    >
      <p>
        daww
      </p>
    </div>
  </div>
</div>
`;
